logicrunner/builtin/contract/migrationadmin/migrationadmin.go

```diff
@@ -0,0 +1,121 @@
+/*
+ *
+ *  Copyright  2019. Insolar Technologies GmbH
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package migrationadmin
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/insolar/insolar/insolar"
+	"github.com/insolar/insolar/logicrunner/builtin/foundation"
+)
+
+// MigrationAdmin manage and change status for  migration daemon.
+type MigrationAdmin struct {
+	foundation.BaseContract
+	MigrationDaemon      foundation.StableMap
+	MigrationAdminMember insolar.Reference
+}
+
+const (
+	StatusActive     = "ACTIVE"
+	StatusInactivate = "INACTIVE"
+)
+
+// Create new Migration admin in genesis.
+func NewMigrationAdmin(migrationDaemons [insolar.GenesisAmountMigrationDaemonMembers]insolar.Reference, migrationAdminMember insolar.Reference) (*MigrationAdmin, error) {
+	daemonMigration := make(foundation.StableMap)
+	for i := 0; i < insolar.GenesisAmountMigrationDaemonMembers; i++ {
+		daemonMigration[migrationDaemons[i].String()] = StatusInactivate
+	}
+	return &MigrationAdmin{MigrationDaemon: daemonMigration, MigrationAdminMember: migrationAdminMember}, nil
+}
+
+// Return stable map migration daemon.
+func (mA MigrationAdmin) GetAllMigrationDaemon() (foundation.StableMap, error) {
+	sizeMap := len(mA.MigrationDaemon)
+	if sizeMap != insolar.GenesisAmountMigrationDaemonMembers {
+		return foundation.StableMap{}, fmt.Errorf(" MigrationAdmin contains the wrong amount migration daemon %d", sizeMap)
+	}
+	return mA.MigrationDaemon, nil
+}
+
+// Activate migration daemon.
+func (mA MigrationAdmin) ActivateDaemon(daemonMember string, caller insolar.Reference) error {
+	if caller == mA.MigrationAdminMember {
```

Comment:
move this condition to separate call that will be immutable

Reply:
you change the status of a demon

Reply:
Read what i wrote again. If you have a question - ask me, please.